sdks/python/src/opik/api_objects/dataset/dataset.py

```diff
@@ -52,6 +55,59 @@ def description(self) -> Optional[str]:
         """The description of the dataset."""
         return self._description
 
+    def _insert_batch_with_retry(
+        self, batch: List[rest_dataset_item.DatasetItemWrite]
+    ) -> None:
+        """Insert a batch of dataset items with automatic retry on rate limit errors."""
+        max_retries = 5
```

Comment:
1. Regular retries are not needed here as they are already configured inside `_rest_client` methods.
2. Let's extract this logic to the following function:
```python
def _ensure_rest_api_call_respecting_rate_limit(
    rest_callable: Callable[[], Any],
):
    while True:
        try:
            result = rest_callable()
            return result
        except ApiError as exception:
                if exception.status_code == 429:
                     # wait for the time in response header, return to the loop and try again
                else:
                     raise